src/components/OSCALComponentDefinitionControlImplementation.js

```diff
@@ -43,32 +45,46 @@ export default function OSCALComponentDefinitionControlImplementation(props) {
             </Grid>
             <Grid item xs={12}>
               <List>
-                {props.controlImplementations.map((controlImpl) => (
-                  <ListItem key={controlImpl.uuid}>
-                    <ListItemText>
-                      {controlImpl.description}
-                      <Grid item xs={12}>
-                        <List
-                          className={
-                            classes.OSCALControlImplementationImplReqList
-                          }
-                        >
-                          {controlImpl["implemented-requirements"].map(
-                            (implementedRequirement) => (
-                              <OSCALControlImplementationImplReq
-                                implementedRequirement={implementedRequirement}
-                                components={props.components}
-                                controls={props.controls}
-                                childLevel={0}
-                                key={implementedRequirement.uuid}
-                              />
-                            )
-                          )}
-                        </List>
-                      </Grid>
-                    </ListItemText>
-                  </ListItem>
-                ))}
+                {props.controlImplementations.map((controlImpl) => {
+                  // Get modifications from source if it is profile
+                  if (!modifications) {
+                    getProfileModifications(
+                      controlImpl.source,
+                      props.parentUrl,
+                      setModifications
+                    );
+                  }
```

Comment:
Maybe this is a larger architectural thing but I think this highlights some questions that I have more largely. So I don't want to hold up this PR but I want to ask to see if there are issues we can create to better get a feel for what's going on here.

So above, we make a call to `useState(props.modifications)` that gives us `modifications` and `setModifications`, both of which we use here. But `getProfileModifications` seems to asynchronously _set_ values on the profile using the passed in `setModifications` parameter (which appears to be a function?). 

We also seem to be passing all these URLs around; I don't see where an actual object gets set/returned. And maybe I don't know the code base well enough, but I feel like like I should get a vibe for where we end up actually getting an object with values somewhere.

I think we've got a few things here:
1. I am expecting this code to look more object-oriented and for a variety of reasons, it isn't and that's on me
2. We probably could a better job naming things. Is `setModifications` setting modifications? Or something else? Both? Is `getProfileModifications` purely getting things? It feels like quite a bit of a _write_ operation.
3. We could probably do a better job of documenting things; but I am also a big believer that names are an important part of documenting so maybe this is the same as 2.

I want to be really clear that I am not complaining about any changes/additions specifically in this code; I realize these are patterns we use in a few places. I _do_ think that there's an opportunity to make things a little better (or at least different) and maybe to identify how we can create better (and more reusable?) patterns elsewhere?